examples/2_advanced/plot_s4_galaxies.py

```diff
@@ -0,0 +1,166 @@
+'''
+Stage IV Space Satellite Galaxy Survey
+======================================
+
+This example simulates a galaxy catalogue from a Stage IV Space Satellite Galaxy Survey such as
+*Euclid* and *Roman* combining the :ref:`sphx_glr_examples_1_basic_plot_density.py` and
+:ref:`sphx_glr_examples_1_basic_plot_lensing.py` examples with generators for
+the intrinsic galaxy ellipticity and the resulting shear with some auxiliary functions.
+
+The focus in this example is mock catalogue generation using auxiliary functions
+built for simulating Stage-IV galaxies.
+'''
+
+# %%
+# Setup
+# -----
+# The basic setup of galaxies and weak lensing fields is the same as in the
+# previous examples.
+#
+# In addition to a generator for intrinsic galaxy ellipticities,
+# following a normal distribution, we also show how to use auxiliary functions
+# to generate photometric redshift distributions and visibility masks.
+#
+# Finally, there is a generator that applies the reduced shear from the lensing
+# maps to the intrinsic ellipticities, producing the galaxy shears.
+
+import numpy as np
+import healpy as hp
+import matplotlib.pyplot as plt
+
+# these are the GLASS imports: cosmology and glass ifself
+from cosmology import LCDM
+from glass import glass
+
+# also needs camb itself to get the parameter object
+import camb
+
+# cosmology for the simulation
+cosmo = LCDM(h=0.7, Om=0.3)
+
+# basic parameters of the simulation
+nside = 512
+lmax = nside
+
+# size of the dz of each shell to integrate along the LoS:
+dz = 0.05
+
+# galaxy density (using 1/100 of the expected galaxy number density for Stage-IV)
+n_arcmin2 = 0.3
+
+# sigma_ellipticity as expected for a stage-IV survey
+sigma_e = 0.27
+
+# photometric redshift error
+sigma_z0 = 0.03
+
+# set up CAMB parameters for matter angular power spectrum
+pars = camb.set_params(H0=100*cosmo.h, omch2=cosmo.Om*cosmo.h**2)
+
+# %%
+# Simulation Setup
+# ----------------
+# Here we setup the overall photometric redshift distribution
+# and separate it into equal density tomographic bins
+# with photometric redshift errors.
```

Comment:
Again prefer not to call it "photometric errors" because we might have a true photometric error model in the future.

```suggestion
# with the typical redshift errors of a photometric survey.
```